Boost >= 1.87 conflict

**Bug summary**
I am using the boost library >= 1.87 in some part of the program that is to be compiled with acpp. Boost is not used within GPU kernels. Compilation fails due to some conflict with:

```
In file included from /home/users/cbyrohl/repos/thor/src/datasets/PointCloudDataset.cpp:21:
In file included from /home/users/cbyrohl/repos/thor/src/datasets/../cgal.h:8:
In file included from /home/users/cbyrohl/repos/thor/extern/cgal/Triangulation_3/include/CGAL/Delaunay_triangulation_3.h:25:
In file included from /home/users/cbyrohl/repos/thor/extern/cgal/Triangulation_3/include/CGAL/Triangulation_3.h:45:
In file included from /home/users/cbyrohl/repos/thor/extern/cgal/Kernel_23/include/CGAL/Bbox_3.h:24:
In file included from /share/software/user/open/boost/1.87.0/include/boost/math/special_functions/next.hpp:20:
/share/software/user/open/boost/1.87.0/include/boost/math/special_functions/fpclassify.hpp:133:19: error: an attribute list cannot appear here
  133 | template<> inline BOOST_MATH_GPU_ENABLED int (fpclassify)(double t)
      |                   ^~~~~~~~~~~~~~~~~~~~~~
/share/software/user/open/boost/1.87.0/include/boost/math/tools/config.hpp:734:58: note: expanded from macro 'BOOST_MATH_GPU_ENABLED'
  734 | #  define BOOST_MATH_GPU_ENABLED BOOST_MATH_CUDA_ENABLED BOOST_MATH_SYCL_ENABLED
      |                                                          ^~~~~~~~~~~~~~~~~~~~~~~
/share/software/user/open/boost/1.87.0/include/boost/math/tools/config.hpp:705:35: note: expanded from macro 'BOOST_MATH_SYCL_ENABLED'
  705 | #  define BOOST_MATH_SYCL_ENABLED SYCL_EXTERNAL
      |                                   ^~~~~~~~~~~~~
/home/users/cbyrohl/bin/../include/AdaptiveCpp/sycl/../hipSYCL/sycl/sycl.hpp:88:25: note: expanded from macro 'SYCL_EXTERNAL'
   88 |   #define SYCL_EXTERNAL [[clang::annotate("hipsycl_sscp_outlining")]]
      |                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

```
This combination leads to invalid C++ attribute positions. Is this something that needs to be addressed in AdaptiveCpp or Boost?

Potentially caused by this: https://github.com/boostorg/math/pull/1161

**Expected behavior**
Same behavior as in < 1.87

**Describe your setup**
Acpp from git 24fe01b3
boost 1.87 from module on HPC cluster

@illuhad, yes replacing above definition does fix the issue at hand. 

Note that the code still does not compile for boost because of another issue:

```
In file included from /home/users/cbyrohl/repos/thor/src/datasets/PointCloudDataset.cpp:21:
In file included from /home/users/cbyrohl/repos/thor/src/datasets/../cgal.h:9:
In file included from /home/users/cbyrohl/repos/thor/extern/cgal/Kernel_23/include/CGAL/Exact_predicates_inexact_constructions_kernel.h:21:
In file included from /home/users/cbyrohl/repos/thor/extern/cgal/Filtered_kernel/include/CGAL/Filtered_kernel.h:22:
In file included from /home/users/cbyrohl/repos/thor/extern/cgal/Kernel_23/include/CGAL/Exact_kernel_selector.h:26:
In file included from /home/users/cbyrohl/repos/thor/extern/cgal/Number_types/include/CGAL/Number_types/internal/Exact_type_selector.h:27:
In file included from /home/users/cbyrohl/repos/thor/extern/cgal/Number_types/include/CGAL/boost_mp.h:42:
In file included from /share/software/user/open/boost/1.87.0/include/boost/multiprecision/number.hpp:12:
In file included from /share/software/user/open/boost/1.87.0/include/boost/multiprecision/detail/generic_interconvert.hpp:12:
In file included from /share/software/user/open/boost/1.87.0/include/boost/multiprecision/detail/default_ops.hpp:2163:
/share/software/user/open/boost/1.87.0/include/boost/multiprecision/detail/functions/pow.hpp:764:38: error: no type named 'rounding_error' in namespace 'boost::math'
  764 |    BOOST_MP_CATCH(const boost::math::rounding_error&)
      |                         ~~~~~~~~~~~~~^
/share/software/user/open/boost/1.87.0/include/boost/multiprecision/detail/no_exceptions_support.hpp:48:42: note: expanded from macro 'BOOST_MP_CATCH'
   48 | #   define BOOST_MP_CATCH(x) BOOST_CATCH(x)
      |                                          ^
/share/software/user/open/boost/1.87.0/include/boost/core/no_exceptions_support.hpp:28:34: note: expanded from macro 'BOOST_CATCH'
   28 | #    define BOOST_CATCH(x) catch(x)
```

I currently fix this by using:

```cpp

#pragma push_macro("SYCL_LANGUAGE_VERSION")
#pragma push_macro("SYCL_EXTERNAL")
#undef SYCL_LANGUAGE_VERSION
#undef SYCL_EXTERNAL

// CGAL/Boost includes (SYCL macros are hidden during these includes)
// ...


// Restore SYCL macros after CGAL/Boost includes
#pragma pop_macro("SYCL_EXTERNAL")
#pragma pop_macro("SYCL_LANGUAGE_VERSION")
```

Any idea how/whether to address this in acpp/boost directly?

--- a/doc/installing.md
+++ b/doc/installing.md
@@ -16,9 +16,10 @@ In order to successfully build and install AdaptiveCpp, the following dependenci
 
 * python 3 (for the `acpp` compiler driver)
 * `cmake`
-* the Boost C++ libraries (in particular, `boost.test` for the unit tests)
+* the Boost C++ libraries (only `boost.test` for the unit tests)
     * it may be helpful to set the `BOOST_ROOT` `cmake` variable to the path to the root directory of Boost you wish to use if `cmake` does not find it automatically
     * **Note for boost 1.78 users:** There seems to be a bug in the build system for boost 1.78, causing the compiled fiber and context libraries not to be copied to the installation directory. You will have to copy these libraries manually to the installation directory. In binary packages from some distribution repositories this issue is fixed. You might be only affected when building boost manually from source.
+    * **Note for boost 1.87 and 1.88 users**: When compiling SYCL applications that use `Boost.Math`, these versions incorrectly enable SYCL-specific code paths, which can cause compilation errors. Use Boost < 1.87 or >= 1.89 instead.
 
 In addition, the various supported [compilation flows](compilation.md) and programming models have additional requirements:
 

